Makefile

```diff
@@ -16,75 +15,141 @@ BRIGADE_GITHUB_GW_EXTERNAL_PORT := 7744
+$1-$2-image: $1/Dockerfile.$2
+	docker build $(DOCKER_BUILD_FLAGS) -t $(DOCKER_REGISTRY)/$1:$2-$(IMAGE_TAG) -f $1/Dockerfile.$2 $1
+
+# Push image for a single arch
+push-$1-$2-image: $1-$2-image
+	docker push $(DOCKER_REGISTRY)/$1:$2-$(IMAGE_TAG)
+
+# Create arch dockerfile
+$1/Dockerfile.$2: $1/Dockerfile
+	@cat $1/Dockerfile | ARCH="$2" QEMU_IMPORT="$$($2_QEMU_IMPORT)" DOCKER_ARCH="$$($2_DOCKER_ARCH)" envsubst '$$$${ARCH} $$$${QEMU_IMPORT} $$$${DOCKER_ARCH}' > $1/Dockerfile.$2
+
+dockerfiles: $1/Dockerfile.$2
+build-$2-images $1-images: $1-$2-image
+push-$2-images push-$1-images: push-$1-$2-image
+endef
+
+define DOCKER_MANIFEST_TARGETS
+.PHONY: create-$1-manifest $1
+
+$1: $1-images
+
+create-$1-manifest: push-$1-images
+	# Remove old manifest if present
+	@if [ -d $(HOME)/.docker/manifests/docker.io_$(DOCKER_REGISTRY)_$1-$(IMAGE_VERSION) ]; then \
+		rm -rf $(HOME)/.docker/manifests/docker.io_$(DOCKER_REGISTRY)_$1-$(IMAGE_VERSION); \
+	fi
+
+	docker manifest create $(DOCKER_REGISTRY)/$1:$(IMAGE_TAG) $(foreach arch,$(ARCHS),$(DOCKER_REGISTRY)/$1:$(arch)-$(IMAGE_TAG))
+	$(foreach arch,$(ARCHS),docker manifest annotate $$($(arch)_DOCKER_ANNOTATIONS) $(DOCKER_REGISTRY)/$1:$(IMAGE_TAG) $(DOCKER_REGISTRY)/$1:$(arch)-$(IMAGE_TAG);)
+	docker manifest push $(DOCKER_REGISTRY)/$1:$(IMAGE_TAG)
+endef
+
+define BRIG_STATIC_ARCH_TARGETS
+.PHONY: build-$3-static
+build: build-$3-static
+
+ifeq ($1,windows)
+ifeq ($2,amd64)
+build-$3-static: bin/$3-$1-$2.exe
+bin/$3-$1-$2.exe: vendor $(shell find $3 -type f -name '*.go')
+	GOOS=$1 GOARCH=$2 CGO_ENABLED=0 go build -ldflags '$(LDFLAGS)' -o ./$$@ ./$3/cmd/$3
+endif
+endif
+
+ifeq ($1,darwin)
+ifeq ($2,amd64)
+build-$3-static: bin/$3-$1-$2
+bin/$3-$1-$2: vendor $(shell find $3 -type f -name '*.go')
+	GOOS=$1 GOARCH=$2 CGO_ENABLED=0 go build -ldflags '$(LDFLAGS)' -o ./$$@ ./$3/cmd/$3
+endif
+endif
+
+ifeq ($1,linux)
+build-$3-static: bin/$3-$1-$2
+bin/$3-$1-$2: vendor $(shell find $3 -type f -name '*.go')
+	GOOS=$1 GOARCH=$$($2_GOARCH) CGO_ENABLED=0 go build -ldflags '$(LDFLAGS)' -o ./$$@ ./$3/cmd/$3
+endif
+endef
+
+define BRIG_STATIC_TARGETS
+$(foreach cmd,$(BINS),$(eval $(call BRIG_STATIC_ARCH_TARGETS,$2,$1,$(cmd))))
+endef
+
+define GO_ARCH_TARGETS
+.PHONY: build-$1 build-$1-images push-$1-images
+build: build-$1
+build-images: build-$1-images
+push-images: push-$1-images
+
+$(foreach os,$(CX_OSES),$(eval $(call BRIG_STATIC_TARGETS,$1,$(os))))
+$(foreach cmd,$(BINS),$(eval $(call GO_CMD_TARGETS,$(cmd),$1)))
+$(foreach img,$(IMAGES),$(eval $(call DOCKER_IMAGE_TARGETS,$(img),$1)))
+endef
+
+$(foreach arch,$(ARCHS),$(eval $(call GO_ARCH_TARGETS,$(arch))))
+$(foreach img,$(IMAGES),$(eval $(call DOCKER_MANIFEST_TARGETS,$(img))))
 
 .PHONY: helm-install
 helm-install: helm-upgrade
 
 .PHONY: helm-upgrade
 helm-upgrade:
 	helm upgrade --install $(BRIGADE_RELEASE) brigade/brigade --namespace $(BRIGADE_NAMESPACE) \
-		--set brigade-github-app.enabled=true \
+		--set gw.service.type=ClusterIP \
```

Comment:
I'd like to go back to how master looks for this target (`gw.*` sets removed in place of `--set brigade-github-app.enabled=true`), as we hope to deprecate the classic GH gateway (`gw`) in favor of the `brigade-github-app` gateway.

Reply:
This was probably just a merge conflict. Pretty sure I didn't intentionally touch the helm-upgrade target.